autogpt_platform/backend/schema.prisma

```diff
@@ -19,45 +19,148 @@ model User {
+
+  @@index([name])
+}
+
 // This model describes the Agent Graph/Flow (Multi Agent System).
-model AgentGraph {
+model Agent {
   id        String    @default(uuid())
   version   Int       @default(1)
   createdAt DateTime  @default(now())
   updatedAt DateTime? @updatedAt
 
   name        String?
   description String?
-  isActive    Boolean @default(true)
-  isTemplate  Boolean @default(false)
 
   // Link to User model
-  userId String
-  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+  createdByUserId String
+  createdByUser   User   @relation(fields: [createdByUserId], references: [id], onDelete: Cascade)
 
-  AgentNodes                  AgentNode[]
-  AgentGraphExecution         AgentGraphExecution[]
-  AgentGraphExecutionSchedule AgentGraphExecutionSchedule[]
+  isGroupAgent Boolean    @default(false)
+  groupId      String?
+  group        UserGroup? @relation(fields: [groupId], references: [id])
+
+  AgentNodes          AgentNode[]
+  AgentGraphExecution AgentGraphExecution[]
 
   // All sub-graphs are defined within this 1-level depth list (even if it's a nested graph).
-  AgentSubGraphs     AgentGraph[] @relation("AgentSubGraph")
+  AgentSubGraphs     Agent[] @relation("AgentSubGraph")
   agentGraphParentId String?
-  AgentGraphParent   AgentGraph?  @relation("AgentSubGraph", fields: [agentGraphParentId, version], references: [id, version], onDelete: Cascade)
+  AgentGraphParent   Agent?  @relation("AgentSubGraph", fields: [agentGraphParentId, version], references: [id, version], onDelete: Cascade)
+
+  ConfiguredAgents            ConfiguredAgent[]
+  WebhookTrigger              WebhookTrigger[]
+  AgentGraphExecutionSchedule AgentGraphExecutionSchedule[]
+  UserAgents                  UserAgents[]
+  UserBlockCredit             UserBlockCredit[]
+  StoreListing                StoreListing[]
+  StoreListingVersion         StoreListingVersion[]
 
   @@id(name: "graphVersionId", [id, version])
+  @@map("AgentGraph")
+}
+
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+////////////////   USER SPECIFIC DATA   ////////////////////
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+
+model ConfiguredAgent {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  name        String
+  description String
+
+  // For agents that can be triggered by webhooks or cronjob
+  // This bool allows use to disable a configured agent without deleting it
+  isActive Boolean @default(true)
+
+  userId String
+  user   User   @relation(fields: [userId], references: [id])
+
+  agentId      String
+  agentVersion Int
+  agent        Agent  @relation(fields: [agentId, agentVersion], references: [id, version])
+
+  Configuration               AgentNodeExecutionInputOutput[] @relation("AgentNodeExecutionInput")
```

Comment:
```suggestion
  InputPresets                AgentNodeExecutionInputOutput[] @relation("AgentNodeExecutionInput")
```

Reply:
Also, (same comment as below) this references `@relation("AgentNodeExecutionInput"` which I don't think is correct.